### ISSUE TITLE
Rule for max-len failing for line under the length limit

### ISSUE THREAD

**Issue by tgrass12** [eslint/eslint#13799]
**Tell us about your environment**

Node version: v14.14.0
npm version: v6.14.8
Local ESLint version: v7.12.0 (Currently used)
Global ESLint version: Not found

**What parser (default, `@babel/eslint-parser`, `@typescript-eslint/parser`, etc.) are you using?**
Whichever is used by `create-react-app` (`@babel/eslint-parser`?).
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```yml
env:
  browser: true
  commonjs: true
  es2021: true
extends:
  - react-app
  - prettier
  - airbnb-base
parserOptions:
  ecmaVersion: 12
ignorePatterns:
  - serviceWorker.js
overrides:
  - files:
      - "*.test.js"
    rules:
      no-unused-vars: 0
rules:
  no-underscore-dangle: 1
  no-plusplus:
    - 1
    - allowForLoopAfterthoughts: true
  import/no-extraneous-dependencies:
    - 2
    - devDependencies:
        - "**/*.test.js"
        - src/setupTests.js

```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
const nutrientTotals = nutrition.mealOccasions.map((mealOccasion) => {
  return mealOccasion.foods.map((food) => {});
});

```

```bash
npx eslint --cache --fix --config .eslintrc.commit.yml .
```

**What did you expect to happen?**
Linter should find no issues
**What actually happened? Please include the actual, raw output from ESLint.**
  `1:1    error  This line has a length of 107. Maximum allowed is 100             max-len`

It seems the rule is reading up to the closing parenthesis after the variable within the inner map, spaces removed.
```js
`const nutrientTotals = nutrition.mealOccasions.map((mealOccasion) => {
return mealOccasion.foods.map((food`.length
// prints 107
```

Running the linter again with a different variable name seems to confirm this as well:
```js
const nutrientTotals = nutrition.mealOccasions.map((mealOccasion) => {
  return mealOccasion.foods.map((someLongerVariable) => {});
});
```
`  1:1    error  This line has a length of 121. Maximum allowed is 100             max-len`
```js
`const nutrientTotals = nutrition.mealOccasions.map((mealOccasion) => {
return mealOccasion.foods.map((someLongerVariable
`.length
// prints 121
```

**Comment by mdjermanovic** [comment]
Hi @tgrass12, thanks for the issue!

I tried the example with your config. 

[arrow-body-style](https://eslint.org/docs/rules/arrow-body-style) first fixes the code to:

```js
const nutrientTotals = nutrition.mealOccasions.map((mealOccasion) => mealOccasion.foods.map((food) => {}));
```

and then `max-len` reports the 107 length.

`max-len` isn't auto-fixable, but you can manually fix the code to:

```js
const nutrientTotals = nutrition.mealOccasions.map(
  (mealOccasion) => mealOccasion.foods.map((food) => {}),
);
```

**Comment by tgrass12** [comment]
Thanks @mdjermanovic.

I'm running prettier right now and it's forcing current format.  I'll just disable the rule for the block for now.

The fix you've shown seems to work as well, however once I add the code within the inner map it fails again. It's similar, hopefully it helps.

```js
  const nutrientTotals = nutrition.mealOccasions
    .map((mealOccasion) =>
      mealOccasion.foods.map((food) =>
        food.nutrients.reduce(getNutrientAmounts, []),
      ),
    )
```


Possibly a separate issue, but `// eslint-disable-next-line` doesn't prevent the rule being assessed in this case, I had to wrap the entire chain between a block:
```
/* eslint-disable max-len */ 
.... 
/* eslint-enable max-len */
```

**Comment by tgrass12** [comment]
The [implicit-arrow-linebreak](https://eslint.org/docs/rules/implicit-arrow-linebreak) rule may also be involved with this issue.

**Comment by mdjermanovic** [comment]
> ```js
>   const nutrientTotals = nutrition.mealOccasions
>     .map((mealOccasion) =>
>       mealOccasion.foods.map((food) =>
>         food.nutrients.reduce(getNutrientAmounts, []),
>       ),
>     )
> ```

If you prefer this formatting, then you'll need either to disable the `implicit-arrow-linebreak` rule, or to reconfigure it to require linebreaks (`"below"` option).

> ```yaml
> extends:
>   - react-app
>   - prettier
>   - airbnb-base
> ```

Are you also using prettier? If so, you should disable all ESLint rules that might cause conflicts, but then `- prettier` ([eslint-config-prettier](https://github.com/prettier/eslint-config-prettier)) should be specified after `- airbnb-base`.

**Comment by tgrass12** [comment]
Thanks, I had missed prettier needed to be at the end.  This has fixed it for me.

**Comment by nzakas** [comment]
Closing, as this issue is resolved.
